File: Euclid/Messages/Incoming/FLATPROPERTYBYITEM.cs
using Euclid.Game;
using Euclid.Messages.Outgoing;
using Euclid.Network.Streams;
using Euclid.Storage.Database.Access;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Euclid.Messages.Incoming
{
    class FLATPROPERTYBYITEM : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {
            var room = player.RoomEntity.Room;

            if (room == null || !room.IsOwner(player.Details.Id))
            {
                return;
            }

            var item = player.Inventory.GetItem(int.Parse(request.GetArgument(2, "/")));

            if (item == null || !item.Definition.Data.IsDecoration)
            {
                return;
            }
            switch (item.Definition.Data.Sprite)
            {
                case "floor":
                    room.Data.Floor = int.Parse(item.Data.CustomData);
                    break;
                case "wallpaper":
                    room.Data.Wallpaper = int.Parse(item.Data.CustomData);
                    break;
            }

            player.Inventory.RemoveItem(item);
            player.Inventory.TurnPage("update");

            RoomDao.SaveRoom(room.Data);
            ItemDao.DeleteItem(item.Data);

            player.Send(new FLATPROPERTY(item.Definition.Data.Sprite, item.Data.CustomData));

        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/FLATPROPERTYBYITEM.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.game.item.Item;
import euclid.game.room.Room;
import euclid.messages.IMessageEvent;
import euclid.messages.outgoing.FLATPROPERTY;
import euclid.network.streams.Request;
import euclid.storage.database.access.ItemDao;
import euclid.storage.database.access.RoomDao;

public class FLATPROPERTYBYITEM implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        Room room = player.getRoomEntity().getRoom();

        if (room == null || !room.isOwner(player.getDetails().getId())) {
            return;
        }

        Item item = player.getInventory().getItem(Integer.parseInt(request.getArgument(2, "/")));

        if (item == null || !item.getDefinition().getData().isDecoration()) {
            return;
        }

        switch (item.getDefinition().getData().getSprite()) {
            case "floor":
                room.getData().setFloor(Integer.parseInt(item.getData().getCustomData()));
                break;
            case "wallpaper":
                room.getData().setWallpaper(Integer.parseInt(item.getData().getCustomData()));
                break;
        }

        player.getInventory().removeItem(item);
        player.getInventory().turnPage("update");

        RoomDao.saveRoom(room.getData());
        ItemDao.deleteItem(item.getData());

        player.send(new FLATPROPERTY(item.getDefinition().getData().getSprite(), item.getData().getCustomData()));
    }
}
